src/Week1/CreditCard.hs

```diff
@@ -0,0 +1,105 @@
+module Week1.CreditCard where
+
+
+{-
+ Exercise 1: We need to first find the digits of a number.
+
+ 1. toDigits should convert positive Integers to a list of digits. (For 0 or
+ negative inputs, toDigits should return the empty list.)
+
+ 2. toDigitsRev should do the same, but with the digits reversed.
+
+ Examples: toDigits 1234 == [1,2,3,4]
+           toDigitsRev 1234 == [4,3,2,1]
+           toDigits 0 == []
+           toDigits (-17) == []
+ -}
+toDigits :: Integer -> [Integer]
+toDigits n
+  | n < 1     = []
+  | otherwise = toDigits (n `div` 10) ++ [n `mod` 10]
+
+
+-- We can perhaps just use reverse function with toDigitsRev' to increase performace
+-- when input integer is very large. This avoids calling ++ recursively.
+toDigits' :: Integer -> [Integer]
+toDigits' n = reverse $ toDigitsRev' n
+
+
+-- A helper function to accumulate the result of digits being taken from an integer
+toDigitsUtil :: [Integer] -> Integer -> [Integer]
+toDigitsUtil xs n
+  | n <= 0 = xs
+  | otherwise = toDigitsUtil ((n `mod` 10) : xs) (n `div` 10) 
+
+-- Another implmenetation of the toDigits function that calls toDigitsUtil with
+-- an empty (accumulator) list as an arg.
+toDigits'' :: Integer -> [Integer]
+toDigits'' n = toDigitsUtil [] n
```

Comment:
This implementation is correct; I just have two comments about style:

(1) The helper function `toDigitsUtil` could be written as part of a `where` clause, since it is only invoked by the `toDigits''` function. Generally, try to write helper functions as part of a where clause (unless they are shared by multiple functions). 

(2) You could switch the order of arguments for `toDigitsUtil`; this way, your function recurses on the first argument, and you can write the recursive case more elegantly, as so: 

`toDigitsUtil (n `div` 10) $ (n `mod` 10) : xs` 

This is more a pointer on style, and you will see that we try to avoid nested parentheses as much as possible in CDMP.

Reply:
@oishikg Done! This makes sense tbh.